webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -0,0 +1,123 @@
+import {EditUserAbstract} from "@mm-modals/edit-user/edit-user.component";
+
+const passwordTester = require('simple-password-tester');
+import {Component} from "@angular/core";
+import {UserSettingsService} from "@mm-services/user-settings.service";
+import {TranslateService} from "@ngx-translate/core";
+import {UpdateUserService} from "@mm-services/update-user.service";
+import {BsModalRef} from "ngx-bootstrap/modal";
+
+const PASSWORD_MINIMUM_LENGTH = 8;
+const PASSWORD_MINIMUM_SCORE = 50;
+
+
+@Component({
+  selector: 'update-password',
+  templateUrl: './update-password.component.html'
+})
+export class UpdatePasswordComponent extends EditUserAbstract {
+
+  editUserModel: {
+    username?,
+    currentPassword?,
+    password?,
+    passwordConfirm?
+  } = {};
+
+  errors: {
+    currentPassword?,
+    password?
+  } = {};
+
+  constructor(
+    bsModalRef: BsModalRef,
+    userSettingsService: UserSettingsService,
+    private translateService: TranslateService,
+    private updateUserService: UpdateUserService,
+  ) {
+    super(bsModalRef, userSettingsService);
+  }
+
+  updatePassword() {
+    this.setProcessing();
+    if (this.validatePasswordFields()) {
+      const updates = { password: this.editUserModel.password };
+      const username = this.editUserModel.username;
+      this.updateUserService.update(username, updates, username, this.editUserModel.currentPassword)
+        .toPromise()
+        .then(() => {
+          this.setFinished();
+          window.location.reload(true);
+        })
+        .catch(err => {
+          if (err.status === -1) { //Offline Status
+            this.translateService.get('online.action.message').toPromise().then(value => {
```

Comment:
It looks like this isn't working as expected. I'm not getting this error to show when I'm disconnected from the net.

Reply:
Yes got the same, looks like the new HTTP client from angular we are using (`HttpClient`) format the HTTP error differently, fixing...